vortex-array/src/array/sparse/compute.rs

```diff
@@ -30,6 +31,14 @@ impl ArrayCompute for SparseArray {
 
 impl AsContiguousFn for SparseArray {
     fn as_contiguous(&self, arrays: &[ArrayRef]) -> VortexResult<ArrayRef> {
+        let all_fill_types_are_equal = arrays
+            .iter()
+            .map(|a| a.as_sparse().fill_value())
+            .all_equal();
+        assert!(
+            all_fill_types_are_equal,
+            "Cannot concatenate SparseArrays with differing fill values"
+        );
```

Comment:
This should be an error not a panic, i.e. wrap it in an if statement and do `vortex_bail!`